src/Plugin/GraphQL/DataProducer/Field/EntityReferenceTrait.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace Drupal\graphql\Plugin\GraphQL\DataProducer\Field;
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\TranslatableInterface;
+
+trait EntityReferenceTrait {
+
+  private function defer($type, $language, $bundles, $access, $accessUser, $accessOperation, $resolver, $context) {
```

Comment:
I'd prefer this to be a protected method and also have a more descriptive name. By separating the function body of this function into smaller reusable chunks we'd also make it more customizable.